dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaInput.java

```diff
@@ -140,12 +145,18 @@ public KafkaInput(KafkaAdapterFactory kafkaAdapterFactory, KafkaConsumer<K, V> k
     }
 
     private Message<V> wrapInMessage(ConsumerRecord<K, V> record) {
+        Message<V> message = null;
         try {
-            return Message.of(record.value(), this.ackTracker.trackRecord(record));
+            Supplier<CompletionStage<Void>> ack = null;
+            if (this.ackTracker != null) {
+                ack = this.ackTracker.trackRecord(record);
+            }
+            message = this.kafkaAdapterFactory.newIncomingKafkaMessage(record, ack);
```

Comment:
It's not valid for `ack` to be `null` (Trying to follow it through, I think `ack` will be passed as the second argument to `Message.of(payload, ack)` which doesn't permit `null`)

Reply:
Reading it through again, it doesn't go to `Message.of()`, you have your own `Message` implementation which accepts `null` as the ack argument.